### PR TITLE
Remove Printing Err When Running Root Command

### LINKED ISSUE #612: subcommand is required Message When Using kn Root Command
## In what area(s)?

/kind good-first-issue
/kind cleanup

## What version of Knative Client?

Master branch

Latest release at this time:

```
Version:      
Build Date:   
Git Revision: 
Supported APIs:
* Serving
  - serving.knative.dev/v1alpha1 (knative-serving v0.11.0)
* Eventing
  - sources.eventing.knative.dev/v1alpha1 (knative-eventing v0.11.0)
  - eventing.knative.dev/v1alpha1 (knative-eventing v0.11.0)
```

When running the root command `kn`, the message `subcommand is required` appears: 

```
Manage your Knative building blocks:

* Serving: Manage your services and release new software to them.
* Eventing: Manage event subscriptions and channels. Connect up event sources.

Usage:
  kn [command]

Available Commands:
  completion  Output shell completion code
  help        Help about any command
  plugin      Plugin command group
  revision    Revision command group
  route       Route command group
  service     Service command group
  source      Event source command group
  trigger     Trigger command group
  version     Prints the client version

Flags:
      --config string        kn config file (default is $HOME/.kn/config.yaml)
  -h, --help                 help for kn
      --kubeconfig string    kubectl config file (default is $HOME/.kube/config)
      --log-http             log http traffic
      --lookup-plugins       look for kn plugins in $PATH
      --plugins-dir string   kn plugins directory (default "~/.kn/plugins")

Use "kn [command] --help" for more information about a command.
subcommand is required
```

When running `kn -h`, the message does not appear. `-h` is not a subcommand, so it's somewhat of an awkward message to communicate to users that a subcommand is required since it's not.

It's a very common use case to run the root command to get a general overview of information about the CLI, so it probably makes sense to not display the message since it seems to imply that something is wrong.

`Use "kn [command] --help" for more information about a command.` should probably be sufficient to inform users about what this message is trying to communicate.

## Steps to Reproduce the Problem

Run `kn`

### PR DESCRIPTION
**Pull request by danielhelfand**
Fixes #612

## Proposed Changes

* Remove printing err when running `kn` root command

### Release

`Fix subcommand is required error from showing when running kn by itself`

### CODE REVIEW

File: cmd/kn/main.go [hunk 1]
```diff
@@ -34,10 +34,12 @@ func main() {
 	defer cleanup()
 	rand.Seed(time.Now().UnixNano())
 	err = core.NewDefaultKnCommand().Execute()
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+	if err.Error() == "subcommand is required" {
 		os.Exit(1)
```
Comment (duglin) [inline cmd/kn/main.go]: Sorry - didn't have time to try this myself (stuck in calls), but if we immediately exit w/o printing anything, what does the user see?

Comment (duglin) [reply cmd/kn/main.go]: I just want to make sure we're not exiting w/o some indication of what went wrong so the user can fix it.

Comment (danielhelfand) [reply cmd/kn/main.go]: So, in the most recent push, the only `Error()` ignored is `subcommand is required`, which is occurring when you run `kn` by itself. So running `kn` will result in the following:

```
Manage your Knative building blocks:

* Serving: Manage your services and release new software to them.
* Eventing: Manage event subscriptions and channels. Connect up event sources.

Usage:
  kn [command]

Available Commands:
  completion  Output shell completion code
  help        Help about any command
  plugin      Plugin command group
  revision    Revision command group
  route       Route command group
  service     Service command group
  source      Event source command group
  trigger     Trigger command group
  version     Prints the client version

Flags:
      --config string        kn config file (default is $HOME/.kn/config.yaml)
  -h, --help                 help for kn
      --kubeconfig string    kubectl config file (default is $HOME/.kube/config)
      --log-http             log http traffic
      --lookup-plugins       look for kn plugins in $PATH
      --plugins-dir string   kn plugins directory (default "~/.kn/plugins")

Use "kn [command] --help" for more information about a command.
```

If the error is not `subcommand is required`, this approach should still print the error as originally implemented.

Comment (duglin) [reply cmd/kn/main.go]: ok thanks.

**Pull request merged**